### PR TITLE
Issue 231 switch syscalls to libaudit

### PR DESCRIPTION
**Pull request by cvonelm**
This PR replaces my handwritten sh script with libaudit for resolving the syscall names.

### CODE REVIEW

**Comment by cvonelm** [comment]
This PR compiles with and without libaudit on aarch64, fixing #231 

libaudit though fails to resolve even simple system call names such as "read" or "write" on this specific aarch64 machine, but this seems to be an issue with the shipped libaudit.

File: CMakeLists.txt [hunk 1]
```diff
@@ -105,12 +104,13 @@ find_package(Doxygen COMPONENTS dot)
 find_package(x86_energy 2.0 CONFIG)
 find_package(StdFilesystem REQUIRED)
 find_package(Sensors)
+find_package(PkgConfig)
 
-GEN_SYSCALLS_HEADER(HEADER_GENERATED)
-
-if(NOT HEADER_GENERATED)
-    message(SEND_ERROR "Could not parse syscall names from <sys/syscall.h>.")
+if(${PkgConfig_FOUND})
```
Comment (bmario) [inline CMakeLists.txt:109]:
```suggestion
if(PkgConfig_FOUND)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -126,6 +126,7 @@ option(UML_LOOK "Generate graphs with an UML look" OFF)
 add_feature_info("USE_RADARE" USE_RADARE "Use Radare to add instruction information to samples.")
 CMAKE_DEPENDENT_OPTION(USE_SENSORS "Use the libsensors to read system metrics." ON "Sensors_FOUND" OFF)
 add_feature_info("USE_SENSORS" USE_SENSORS "Use the libsensors to read system metrics.")
+CMAKE_DEPENDENT_OPTION(USE_LIBAUDIT "Use libaudit for syscall name resolution" ON Audit_FOUND OFF)
```
Comment (bmario) [inline CMakeLists.txt:129]: Use `add_feature_info` to add syscall resolution to the feature list

File: src/perf/time/reader.cpp [hunk 3]
```diff
@@ -92,9 +92,11 @@ Reader::Reader()
     catch (...)
     {
 #ifndef USE_HW_BREAKPOINT_COMPAT
-        Log::error() << "time synchronization with hardware breakpoints failed, try rebuilding lo2s with -DUSE_HW_BREAKPOINT_COMPAT=ON";
+        Log::error() << "time synchronization with hardware breakpoints failed, try rebuilding "
```
Comment (bmario) [inline src/perf/time/reader.cpp:95]: Unrelated reformat

File: src/monitor/process_monitor.cpp [hunk 4]
```diff
@@ -55,8 +55,9 @@ void ProcessMonitor::insert_thread(Process process, Thread thread, std::string n
         perf::counter::CounterProvider::instance().has_group_counters(ExecutionScope(thread)) ||
         perf::counter::CounterProvider::instance().has_userspace_counters(ExecutionScope(thread)))
     {
-        auto inserted = threads_.emplace(std::piecewise_construct, std::forward_as_tuple(thread),
-                                         std::forward_as_tuple(ExecutionScope(thread), *this, spawn));
+        auto inserted =
+            threads_.emplace(std::piecewise_construct, std::forward_as_tuple(thread),
```
Comment (bmario) [inline src/monitor/process_monitor.cpp:59]: unrelated reformat

File: src/monitor/cpu_set_monitor.cpp [hunk 5]
```diff
@@ -72,21 +72,23 @@ CpuSetMonitor::CpuSetMonitor() : MainMonitor()
         {
             Log::debug() << "Create cstate recorder for cpu #" << cpu.as_int();
 
-            auto inserted = monitors_.emplace(std::piecewise_construct, std::forward_as_tuple(cpu),
-                                            std::forward_as_tuple(ExecutionScope(cpu), *this, false));
+            auto inserted =
```
Comment (bmario) [inline src/monitor/cpu_set_monitor.cpp:75]: unrelated reformat

**Review by bmario — CHANGES_REQUESTED** [review]
I don't like that the lib audit code is creeping into config.cpp and trace.cpp. Please abstract that out

**Pull request merged**